app/src/main/java/com/haruta/harutyan/originalapp/AddLocationActivity.kt

```diff
@@ -46,16 +51,25 @@ class AddLocationActivity : AppCompatActivity(),
         db = AppDatabase.getInstance(this.applicationContext)!!
 
         binding.saveFab.setOnClickListener {
-            //保存するデータの変数を作成
-            val location: Location = Location(
-                name = "",
-                latitude = latitude,
-                longitude = longitude,
-            )
-            //Daoのinsertを呼び出して保存したいUserを渡す
-            db.locationDao().insert(location)
-
-            finish()
+            name = binding.locationNameTextEdit.text.toString()
+
+            AlertDialog.Builder(this)
+                .setTitle("${name}を登録します")
+                .setPositiveButton("OK"){ dialog, which ->
+                    //保存するデータの変数を作成
+                    val location: Location = Location(
+                        name = name,
+                        latitude = latitude,
+                        longitude = longitude,
+                    )
+                    //Daoのinsertを呼び出して保存したいUserを渡す
+                    db.locationDao().insert(location)
+                    finish()
+                }
+                .setNegativeButton("キャンセル"){  dialog, which ->
+
+                }
```

Comment:
細かいけど、引数を使わない場合はこうやって書けるよ！

```suggestion
                .setNegativeButton("キャンセル"){  _, _ -> }
```